panic: runtime error: invalid memory address or nil pointer dereference

- CentOS 6.5
- Telegraf v1.2.1 (git: release-1.2 3b6ffb344e5c03c1595d862282a6823ecb438cff)
- Influxdb version: 1.2.0
- Kapacitor version 1.2.0, branch master, commit 5408057e5a3493d3b5bd38d5d535ea45b587f8ff


I am having a problem when trying to start kapacitor, start/restart commands fail.
`service kapacitor start` says it has started [OK] , but `service kapacitor status` says [FAILED]

When checking _/var/log/kapacitor/kapacitor.err_ , I get the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x51c319]

goroutine 1 [running]:
panic(0xb6fa40, 0xc420010080)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/influxdata/kapacitor/server.(*Server).setupIDs(0xc4201ba180, 0xc4201ad858, 0x2)
        /root/go/src/github.com/influxdata/kapacitor/server/server.go:715 +0x79
github.com/influxdata/kapacitor/server.New(0xc420142080, 0xccd314, 0x5, 0xcd31c0, 0x28, 0xccd31c, 0x6, 0x10ae1a0, 0xc42000d7c0, 0x0, ...)
        /root/go/src/github.com/influxdata/kapacitor/server/server.go:142 +0x55f
github.com/influxdata/kapacitor/cmd/kapacitord/run.(*Command).Run(0xc42000ebd0, 0xc42000c0b0, 0x4, 0x4, 0xc420089901, 0x108)
        /root/go/src/github.com/influxdata/kapacitor/cmd/kapacitord/run/command.go:121 +0xa81
main.(*Main).Run(0xc4201adef0, 0xc42000c0b0, 0x4, 0x4, 0xc420043f38, 0x40740b)
        /root/go/src/github.com/influxdata/kapacitor/cmd/kapacitord/main.go:80 +0x2e2
main.main()
        /root/go/src/github.com/influxdata/kapacitor/cmd/kapacitord/main.go:41 +0x20b
```

`$kapacitord `runs fine and starts kapactiro. But I prefer to use `service start kapacitor`

I understand that similar issues have been raised and fixed before ( like here: https://github.com/influxdata/kapacitor/issues/1087).

Is the fix included in the version I am running or I should build the binaries myself?

Thanks

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -84,6 +84,7 @@ kapacitor define-handler system aggregate_by_1m.yaml
 - [#1295](https://github.com/influxdata/kapacitor/issues/1295): Fix curruption of recordings list after deleting all recordings.
 - [#1237](https://github.com/influxdata/kapacitor/issues/1237): Fix missing "vars" key when listing tasks.
 - [#1271](https://github.com/influxdata/kapacitor/issues/1271): Fix bug where aggregates would not be able to change type.
+- [#1261](https://github.com/influxdata/kapacitor/issues/1261): Fix panic when the process cannot stat the data dir.
 
 ## v1.2.0 [2017-01-23]
 

--- a/server/server.go
+++ b/server/server.go
@@ -738,9 +738,13 @@ func (s *Server) Close() error {
 
 func (s *Server) setupIDs() error {
 	// Create the data dir if not exists
-	if f, err := os.Stat(s.dataDir); err != nil && os.IsNotExist(err) {
-		if err := os.Mkdir(s.dataDir, 0755); err != nil {
-			return errors.Wrapf(err, "data_dir %s does not exist, failed to create it", s.dataDir)
+	if f, err := os.Stat(s.dataDir); err != nil {
+		if os.IsNotExist(err) {
+			if err := os.Mkdir(s.dataDir, 0755); err != nil {
+				return errors.Wrapf(err, "data_dir %q does not exist, failed to create it", s.dataDir)
+			}
+		} else {
+			return errors.Wrapf(err, "failed to stat data dir %q", s.dataDir)
 		}
 	} else if !f.IsDir() {
 		return fmt.Errorf("path data_dir %s exists and is not a directory", s.dataDir)

